### PR TITLE
adjustments for oranda:

### PR DESCRIPTION
**Pull request by Gankra**
* swap the order of the start_dir and clamp_dir
* add the manifest path to the broken case
* expose find_file in the public API

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -187,9 +192,10 @@ impl JsonWorkspaceSearch {
             WorkspaceSearch::Found(info) => {
                 JsonWorkspaceSearch::Found(JsonWorkspaceInfo::from_real(root, info))
             }
-            WorkspaceSearch::Broken(e) => {
-                JsonWorkspaceSearch::Broken(json_diagnostic_from_real(root, e))
-            }
+            WorkspaceSearch::Broken {
+                manifest_path: _,
```
Comment (ashleygwilliams) [inline src/main.rs:196]: why do we drop the manifest path here?

Comment (Gankra) [reply src/main.rs]: good call, fixed

File: src/main.rs [hunk 2]
```diff
@@ -53,8 +53,13 @@ fn print_search(search: WorkspaceSearch, name: &str) {
...
             eprintln!("{:?}", err);
         }
-        WorkspaceSearch::Broken(e) => {
-            let err = Report::new(e).wrap_err(format!("Your {name} project has some issues"));
+        WorkspaceSearch::Broken {
+            manifest_path,
+            cause,
+        } => {
+            let err = Report::new(cause).wrap_err(format!(
+                "We found a broken {name} project at {manifest_path}"
```
[the remark this answers is not in the document]
Comment (Gankra) [reply src/main.rs]: Went with "We found a potential {name} project at {manifest_path}, but there was an issue"

[the remark this answers is not in the document]
Comment (ashleygwilliams) [reply src/main.rs]: oh nice i like that even better than mine

**Review by ashleygwilliams — APPROVED** [review]

**Pull request merged**